language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/AbstractCodeActionTest.java

```diff
@@ -132,8 +137,33 @@ public void test(String config) throws IOException, WorkspaceDocumentException {
 
                 // Match edits
                 if (expected.edits != null) {
-                    JsonElement editsElement = right.get("edit").getAsJsonObject().get("documentChanges")
-                            .getAsJsonArray().get(0).getAsJsonObject().get("edits");
+                    JsonElement editsElement;
+                    if (expected.resolvable) {
+                        Assert.assertFalse(right.has("edit"));
+                        ResolvableCodeAction action = ResolvableCodeAction.from(gson.fromJson(right, CodeAction.class));
+                        String response = TestUtil.getCodeActionResolveResponse(getServiceEndpoint(), action);
+                        JsonObject resolvedResponse = getResponseJson(response);
+                        editsElement = resolvedResponse.get("result").getAsJsonObject().get("edit").getAsJsonObject()
+                                .get("documentChanges").getAsJsonArray().get(0).getAsJsonObject().get("edits");
+                        JsonElement actualData = right.get("data");
+                        if (expected.data == null) {
+                            if (!actualData.isJsonNull()) {
+                                this.alterFileUri(actual, actualData, sourceRoot);
+                            }
+
+                        } else {
+                            this.alterExpectedUri(expected.data, sourceRoot);
+                            JsonElement expectedData = gson.toJsonTree(expected.data);
+                            if (!expectedData.equals(actualData)) {
+                                this.alterFileUri(actual, actualData, sourceRoot);
+                            }
+                        }
```

Comment:
We can simplify this logic as follows.
```suggestion
                        if(!actualData.isJsonNull()) {
                               this.alterFileUri(actual, actualData, sourceRoot);
                        }
                        if (expected.data != null) {
                              this.alterExpectedUri(expected.data, sourceRoot);
                             JsonElement expectedData = gson.toJsonTree(expected.data);
                        }
```